File: js/structure-in-rte/ImplementResolver.js
// DocSection: structure_in_rte_implement_resolver
// This is a global resolver for all tweets. You can also define a resolver for a given query: https://docs.kontent.ai/js-resolve-query
class Tweet extends ContentItem {
    constructor() {
        super({
            richTextResolver: (tweet) => {
                return `<blockquote class='twitter-tweet' data-lang='en' data-theme='`+tweet.theme.value[0].codename+`'><a href='`+tweet.tweet_link.value+`'></a></blockquote>`;
            }
        })
    }
}

var typeResolvers = [
    new TypeResolver('tweet', (rawData) => new Tweet())
];
// EndDocSection

File: ruby/structure-in-rte/ImplementResolver.rb
# DocSection: structure_in_rte_implement_resolver
require 'delivery-sdk-ruby'

item_resolver = Kentico::Kontent::Delivery::Resolvers::InlineContentItemResolver.new(lambda do |item|
  if item.system.type.eql? 'tweet'
    return "<blockquote class=\"twitter-tweet\" data-lang=\"en\" data-theme=\"#{item.elements.theme.value[0].codename}\">"\
    "<a href=\"#{item.elements.tweet_link.value}\"></a>"\
    '</blockquote>'
  end
end)
# EndDocSection